packets1/auth_test.go

```diff
@@ -1,13 +1,117 @@
 package packets1
 
 import (
+	"bytes"
+	"fmt"
+	"reflect"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+
+	pkts "github.com/energomonitor/bisquitt/packets"
 )
 
-func TestAuthMarshal(t *testing.T) {
+func TestAuthPlainConstructor(t *testing.T) {
+	assert := assert.New(t)
+
+	user := "test-user"
+	password := []byte("test-password")
+	pkt := NewAuthPlain(user, password)
+
+	if pkt == nil {
+		t.Fatal("New packet should not be nil")
+	}
+
+	assert.Equal("*packets1.Auth", reflect.TypeOf(pkt).String(), "Type should be Connect")
+	assert.Equal(uint8(0), pkt.Reason, "Bad Reason value")
+	assert.Equal("PLAIN", pkt.Method, "Bad Method value")
+	assert.Equal(
+		[]byte(fmt.Sprintf("\x00%s\x00%s", user, password)),
+		pkt.Data, "Bad Data value")
+}
```

Comment:
In tests of other packets you also test `PacketLength()` here. Any reason why not for this packet?

Reply:
It's a very old code. From 29 packets, only 11 are tested for right `PacketLength()`. From a quick view, I suppose these are precisely the fixed-length packets.

Because we now have `Test*Marshal` and `Test*UnmarshalInvalid` tests, this packet length test is probably not needed. Anyway, I'd rather fix it this way or another in a separate PR for all packet types.

Reply:
> Anyway, I'd rather fix it this way or another in a separate PR for all packet types.

Fine with me.